cmd/collector/app/handler/grpc_handler.go

```diff
@@ -60,3 +69,23 @@ func (g *GRPCHandler) PostSpans(ctx context.Context, r *api_v2.PostSpansRequest)
 	}
 	return &api_v2.PostSpansResponse{}, nil
 }
+
+// WithTenant creates a Context with a tenant association
+func (g *GRPCHandler) getTenant(ctx context.Context) (string, error) {
+	// If the collector isn't looking for a tenant, no tenant is needed
+	if g.tenantHeader == "" {
+		return "", nil
+	}
+	md, ok := metadata.FromIncomingContext(ctx)
+	if !ok {
+		return "", status.Errorf(codes.Internal, "missing metadata")
```

Comment:
is it an error to be missing metadata?

Reply:
fair enough, but then I would not treat it as `codes.Internal`, but similar to other ones and I would alter the error message, e.g. "missing metadata, tenant header is required"